lib/protoparser/prometheus/parser.go

```diff
@@ -181,7 +274,68 @@ func (r *Row) unmarshal(s string, tagsPool []Tag, noEscapes bool) ([]Tag, error)
 		// See https://github.com/OpenObservability/OpenMetrics/blob/master/specification/OpenMetrics.md#timestamps
 		ts *= 1000
 	}
-	r.Timestamp = int64(ts)
+	tvt.Timestamp = int64(ts)
+	return tagsPool, tvt, nil
+}
+
+// Returns possible comments that could be exemplars
+func (r *Row) unmarshalMetric(s string, tagsPool []Tag, noEscapes bool) (string, []Tag, error) {
+	tagsStart := len(tagsPool)
+	var err error
+	var tvt *tagsValueTimestamp
```

Comment:
It's better to acquire and release pool objects at the same place:

```go
tvt := getTVT()
defer putTVT(tvt)
if err := tvt.parse(s, tagsPool, noEscapes); err != nil {....}
```

Also, you cannot refer pooled object fields, after release. In this case, it should be fine, but in general, it's better to store it at variable, e.g. `tailComments := tvt.Comments` and return it. Of course, it's copy primitive.